guides/05_custom-components/05_frontend.md

```diff
@@ -0,0 +1,227 @@
+
+>💡 Tip: Gradio components use svelte. Writing svelte is fun! If you're not familiar, check out the interactive [guide](https://learn.svelte.dev/tutorial/welcome-to-svelte)
+
+## The directory structure 
+
+The frontend code should have at the minimum three files:
+
+* `Index.svelte`: This is the main export and where your component's layout and logic should live.
+* `Example.svelte`: This is where the example view of the component is defined.
+
+Feel free to add additional files and subdirectories. 
+If you want to export any additional modules, remember to modify the `package.json` file
+
+```json
+"exports": {
+    ".": "./Index.svelte",
+    "./example": "./Example.svelte",
+    "./package.json": "./package.json"
+},
+```
+
+## The Index.svelte file
+
+Your component should expose the following props that will be passed down from the parent Gradio application.
+
+```
+import type { LoadingStatus } from "@gradio/statustracker";
+import type { Gradio } from "@gradio/utils";
+
+export let gradio: Gradio<{
+    event_1: never;
+    event_2: never;
+}>;
+
+export let elem_id = "";
+export let elem_classes: string[] = [];
+export let scale: number | null = null;
+export let min_width: number | undefined = undefined;
+export let loading_status: LoadingStatus | undefined = undefined;
+export let mode: "static" | "interactive";
+```
+
+* `elem_id` and `elem_classes` allow Gradio app developers to target your component with custom css and javascript from the python `Blocks` class.
+
+* `scale` and `min_width` allow Gradio app developers to control how much space your component takes up in the UI.
+
+* `loading_status` is used to display a loading status over the component when it is the output of an event.
+
+* `mode` is how the parent Gradio app tells your component whether the `interactive` or `static` version should be displayed.
+
+* `gradio`: The gradio object is created by the parent Gradio app. It stores some application-level configuration that will be useful in your component, like internationalization. You must use it to dispach events from your component.
+
+A minimal `Index.svelte` file would look like:
+
+```
+<script lang="ts">
+	import type { LoadingStatus } from "@gradio/statustracker";
+    import { Block } from "@gradio/atoms";
+	import { StatusTracker } from "@gradio/statustracker";
+	import type { Gradio } from "@gradio/utils";
+
+	export let gradio: Gradio<{
+		event_1: never;
+		event_2: never;
+	}>;
+
+    export let value = "";
+	export let elem_id = "";
+	export let elem_classes: string[] = [];
+	export let scale: number | null = null;
+	export let min_width: number | undefined = undefined;
+	export let loading_status: LoadingStatus | undefined = undefined;
+    export let mode: "static" | "interactive";
+</script>
+
+<Block
+	visible={true}
+	{elem_id}
+	{elem_classes}
+	{scale}
+	{min_width}
+	allow_overflow={false}
+	padding={true}
+>
+	{#if loading_status}
```

Comment:
It might be worth explaining why everything uses a `Block` and what the `StatusTracker` is. Can anyone take the lead here?